src/components/app/App.js

```diff
@@ -87,7 +87,21 @@ class App extends Component {
 
   // todo 1: добавить здесь функцию onUserAdd
   //  она должна добавлять пользователя в список users в стейте
-  //  при добавлении пользователя ему нужно добавить пропертю id, можно по аналогии со строкой 82
+  //  при добавлении пользователя ему нужно добавить пропертю id, можно по аналогии со строкой 82 +
+  onUserAdd = (firstName, lastName, email, address) => {
+    const newUsers  = {
+      first_name:firstName,
```

Comment:
If you create object here, you also can add `id` right into the object, so you will not need to spread it on lines 100-101

Reply:
spread/rest operator is only needed when you need to create new object 'on-a-flight'
and here you create new object on lines 92-97